diff --git a/config.def.h b/config.def.h
index 1edb647..dd3eb31 100644
--- a/config.def.h
+++ b/config.def.h
@@ -21,3 +21,6 @@ static unsigned int lines      = 0;
  * for example: " /?\"&[]"
  */
 static const char worddelimiters[] = " ";
+
+/* Size of the window border */
+static unsigned int border_width = 0;
diff --git a/dmenu.c b/dmenu.c
index 27b7a30..7c130fc 100644
--- a/dmenu.c
+++ b/dmenu.c
@@ -685,11 +709,13 @@ setup(void)
        swa.override_redirect = True;
        swa.background_pixel = scheme[SchemeNorm][ColBg].pixel;
        swa.event_mask = ExposureMask | KeyPressMask | VisibilityChangeMask;
-       win = XCreateWindow(dpy, root, x, y, mw, mh, 0,
+       win = XCreateWindow(dpy, root, x, y, mw, mh, border_width,
                            CopyFromParent, CopyFromParent, CopyFromParent,
                            CWOverrideRedirect | CWBackPixel | CWEventMask, &swa);
-       XSetClassHint(dpy, win, &ch);
+       if (border_width)
+               XSetWindowBorder(dpy, win, scheme[SchemeSel][ColBg].pixel);

+       XSetClassHint(dpy, win, &ch);

        /* input methods */
        if ((xim = XOpenIM(dpy, NULL, NULL, NULL)) == NULL)
@@ -716,7 +742,7 @@ setup(void)
 static void
 usage(void)
 {
-       die("usage: dmenu [-bfiv] [-l lines] [-p prompt] [-fn font] [-m monitor]\n"
+       die("usage: dmenu [-bcfiv] [-l lines] [-p prompt] [-bw size] [-fn font] [-m monitor]\n"
            "             [-nb color] [-nf color] [-sb color] [-sf color] [-w windowid]");
 }

@@ -735,6 +761,8 @@ main(int argc, char *argv[])
                        topbar = 0;
                else if (!strcmp(argv[i], "-f"))   /* grabs keyboard before reading stdin */
                        fast = 1;
+               else if (!strcmp(argv[i], "-c"))   /* centers dmenu on screen */
+                       centered = 1;
                else if (!strcmp(argv[i], "-i")) { /* case-insensitive item matching */
                        fstrncmp = strncasecmp;
                        fstrstr = cistrstr;
@@ -747,6 +775,8 @@ main(int argc, char *argv[])
                        mon = atoi(argv[++i]);
                else if (!strcmp(argv[i], "-p"))   /* adds prompt to left of input field */
                        prompt = argv[++i];
+               else if (!strcmp(argv[i], "-bw"))  /* border width */
+                       border_width = atoi(argv[++i]);
                else if (!strcmp(argv[i], "-fn"))  /* font or font set */
                        fonts[0] = argv[++i];
                else if (!strcmp(argv[i], "-nb"))  /* normal background color */
